test/internet/test-http-dns-fail.js

```diff
@@ -19,9 +18,7 @@ function httpreq(count) {
     port: 80,
     path: '/',
     method: 'GET'
-  }, function(res) {
-    resDespiteError = true;
-  });
+  }, common.fail);
```

Comment:
This pattern produces _mildly_ better errors than `assert.fail`...

We should probably have a `common.failWith(string)` that looks like:

```js
function failWith(string) {
  return function failed() {
    assert.fail(null, null, string)
  }
}
```